### PR TITLE
Add commits.yml generation stage before builds

### LINKED ISSUE #2325: [Enhancement] Provide commit id information for each repo before distribution build starts
[Enhancement] Provide commit id information for each repo before distribution build starts.

Return a link to the yml file with all the commit-ids of repos from the input manifest file.

### PR DESCRIPTION
**Pull request by peterzhuamazon**
### Description
Add commits.yml generation stage before builds.
Related docker images: https://hub.docker.com/r/opensearchstaging/ci-runner/tags?page=1&name=ci-runner-alpine3-yq-v1

https://ci-staging.opensearch.org/job/zhujiaxi/job/bundle-build-zhujiaxi/30/console
https://ci-staging.opensearch.org/job/zhujiaxi/job/bundle-build-zhujiaxi/31/console

### Issues Resolved
#2325

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by prudhvigodithi** [comment]
Thanks for this PR @peterzhuamazon 
I see lot of linux parsing inside groovy `vars/getRepositoryCommit.groovy`, is there a way we can do natively with groovy ?

**Comment by peterzhuamazon** [comment]
> Thanks for this PR @peterzhuamazon I see lot of linux parsing inside groovy `vars/getRepositoryCommit.groovy`, is there a way we can do natively with groovy ?

It is actually more stable and easy to do this through shell than using groovy as the Jenkins version of groovy is very unreliable on these things.

We are running this on alpine3 anyway so there is no need to make it general on multiple runners.

Thanks.

**Comment by gaiksaya** [comment]
> Thanks for this PR @peterzhuamazon I see lot of linux parsing inside groovy `vars/getRepositoryCommit.groovy`, is there a way we can do natively with groovy ?

I agree. We can use groovy code to make it maintainable and testable. Shell is always hard to parse through and maintain in long term.

**Review by prudhvigodithi — APPROVED** [review]

**Comment by peterzhuamazon** [comment]
> > Thanks for this PR @peterzhuamazon I see lot of linux parsing inside groovy `vars/getRepositoryCommit.groovy`, is there a way we can do natively with groovy ?
> 
> I agree. We can use groovy code to make it maintainable and testable. Shell is always hard to parse through and maintain in long term.

I will move the shell to a different file and restructure this library with new methods in a improve PR.
Thanks.

**Comment by prudhvigodithi** [comment]
The shell script can be migrated to `scripts/` folder and can be called from there, approving it to start with something and improve in next PR @peterzhuamazon 
Thank you
